src/packages/solid/components/Button/Button.tsx

```diff
@@ -52,25 +57,27 @@ const Button: Component<ButtonProps> = props => {
       forwardStates
     >
       <View forwardStates style={styles.FlexContainer} width={props.width}>
+
+        <Show when={props.prefix?.src}>
+          <Icon {...prefixIcon} />
+        </Show>
+
         <Show when={props.prefix?.icon}>
```

Comment:
What if we had props.checkBox and props.icon and then props.prefix & suffix as booleans to enable showing them

Reply:
it's possible to have checkbox, icon, or both for both prefix and suffix, so we'd still need a way to describe that, and provide potentially different props for each of those prefixes/suffixes